File: 7KYU/Count up the points for the 7 Wonders board game! Easy version/solution.c
unsigned int min(unsigned int a, unsigned int b, unsigned int c) {
  if(a <= b && a <= c) {
    return a;
  } else if (b <= a && b <= c) {
    return b;
  }
  return c;
}

unsigned int seven_wonders_science(unsigned int compasses, unsigned int gears, unsigned int tablets) {
  return compasses * compasses +
         gears * gears +
         tablets * tablets +
         min(compasses, gears, tablets) * 7;
}

File: 7KYU/Count up the points for the 7 Wonders board game! Easy version/solution.go
package kata

import (
	"math"
)

// Calculate how many points you would make in 7 Wonders from these science cards!
func SevenWondersScience(compasses int, gears int, tablets int) int {
	println(compasses)
	println(gears)
	println(tablets)
	return (compasses * compasses) +
		(gears * gears) +
		(tablets * tablets) +
		int((math.Min(math.Min(float64(compasses), float64(gears)), float64(tablets)) * 7))
}
